cura/ExtrudersModel.py

```diff
@@ -63,6 +68,25 @@ def setAddGlobal(self, add):
     def addGlobal(self):
         return self._add_global
 
+    def _onActiveExtruderChanged(self):
+        manager = cura.ExtruderManager.ExtruderManager.getInstance()
+        active_extruder_stack = manager.getActiveExtruderStack()
+        if self._active_extruder_stack != active_extruder_stack:
+            if self._active_extruder_stack:
+                self._active_extruder_stack.containersChanged.disconnect(self._onExtruderStackContainersChanged)
+
+            if active_extruder_stack:
+                # Update the model when the material container is changed
+                active_extruder_stack.containersChanged.connect(self._onExtruderStackContainersChanged)
+            self._active_extruder_stack = active_extruder_stack
+
+
+    def _onExtruderStackContainersChanged(self, container):
+        if container.getMetaDataEntry("type") == "material":
```

Comment:
Yes. Apparently this need documentation ;-)

I'm only interested in changes in material here, because the material-name, and the material-color are used in multiple places in the interface (everywhere where an extruder is selected).